tests/test_gpg.py

```diff
@@ -778,5 +779,24 @@ def test_gpg_sign_and_verify_object_with_specific_key(self):
     self.assertFalse(verify_signature(signature, key_data, wrong_data))
 
 
+  def test_verify_short_signature(self):
+    """Correctly verify a special-crafted short signature. """
+
+    test_data = b"hello"
+    signature_path = os.path.join(self.gnupg_home, "short.sig")
+
+    # Read special-crafted raw gpg signature that is one byte too short
+    with open(signature_path, "rb") as f:
+      signature_data = f.read()
+
+    # Check that the signature is padded upon parsing
+    signature = parse_signature_packet(signature_data)
+    self.assertTrue(len(signature["signature"]) == (ED25519_SIG_LENGTH * 2))
```

Comment:
So this is double length here because signature is already a hex string at this point?

Reply:
Exactly. I'll add a comment.